cluster/src/main/java/org/apache/iotdb/cluster/client/sync/SyncDataClient.java

```diff
@@ -19,64 +19,142 @@
 
 package org.apache.iotdb.cluster.client.sync;
 
+import org.apache.iotdb.cluster.client.ClientCategory;
+import org.apache.iotdb.cluster.client.IClientPool;
 import org.apache.iotdb.cluster.config.ClusterConstant;
 import org.apache.iotdb.cluster.rpc.thrift.Node;
+import org.apache.iotdb.cluster.rpc.thrift.TSDataService.Client;
+import org.apache.iotdb.cluster.utils.ClientUtils;
 import org.apache.iotdb.db.utils.TestOnly;
+import org.apache.iotdb.rpc.RpcTransportFactory;
+import org.apache.iotdb.rpc.TConfigurationConst;
+import org.apache.iotdb.rpc.TimeoutChangeableTransport;
 
+import org.apache.commons.pool2.KeyedPooledObjectFactory;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
 import org.apache.thrift.protocol.TProtocol;
 import org.apache.thrift.protocol.TProtocolFactory;
+import org.apache.thrift.transport.TSocket;
 import org.apache.thrift.transport.TTransportException;
 
+import java.net.SocketException;
+
 /**
  * Notice: Because a client will be returned to a pool immediately after a successful request, you
- * should not cache it anywhere else or there may be conflicts.
+ * should not cache it anywhere else.
  */
-// the two classes does not share a common parent and Java does not allow multiple extension
-@SuppressWarnings("common-java:DuplicatedBlocks")
-public class SyncDataClient extends TSDataServiceClient {
+// TODO: Refine the interfaces of TSDataService. TSDataService interfaces doesn't need extends
+// TODO: RaftService interfaces
+public class SyncDataClient extends Client {
```

Comment:
Since you have done a great abstraction, can we merge `SyncDataClient` and `SyncMetaClient` into one class? It seems their difference only lies on `category` now?

Reply:
I'd really like to do that. But DataClient and MetaClient need to extend different super class separately: TSDataService.(Async)Client and TSMetaService.(Async)Client. Another way is make Client is a wrapper of  TSDataService.(Async)Client and TSMetaService.(Async)Client. It's means we need to rewrite all methods in (Async)Client which I don't think a good way. Actually, we should reorg the client interfaces to make it much clear but I don't suggest we do in this refactor as it has been a big change.

Reply:
I see! These two `Client` are different! Then I'm ok with that. We can talk about refactoring it later.